x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { CoreStart } from 'kibana/public';
+import { AlertData } from '../../../../../common/types';
+import { GlobalState } from '../reducer';
+import { AppAction } from '../action';
+
+// TODO, move this somewhere
+type MiddlewareFactory = (
+  coreStart: CoreStart
+) => (
+  api: MiddlewareAPI<Dispatch<AppAction>, GlobalState>
+) => (next: Dispatch<AppAction>) => (action: AppAction) => unknown;
+
+export const alertMiddlewareFactory: MiddlewareFactory = coreStart => {
+  return store => next => async (action: AppAction) => {
+    next(action);
+    if (action.type === 'userNavigatedToPage' && action.payload === 'alertsPage') {
```

Comment:
I'm assuming by your above `TODO`, that all of this is still moving code for now, but would we be to define these action types in a `types.ts` file, so they're easy to use anywhere?

Reply:
@alexk307 We define actions in files like this one https://github.com/elastic/kibana/pull/55800/files#diff-c971622c4d5a45ad624ef764513a0131. The type checker will throw and error if you use an invalid action type in this if-statement. So if we were to do `if (action.type === 'fooAction')`, the type checker will throw an error because `fooAction` is not of type `AppAction`.